digitaltwin_client/src/dt_client_core.c

```diff
@@ -668,19 +668,16 @@ static DT_SEND_TELEMETRY_CALLBACK_CONTEXT* CreateDTSendTelemetryCallbackContext(
 
 
 // RegisterDTInterfaces_Callback is called when we receive acknowledgement from the service that the message we sent registering the interfaces was received.
-static void RegisterDTInterfaces_Callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void* userContextCallback)
+static void RegisterDTInterfaces_Callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, DT_CLIENT_CORE* dtClientCore)
```

Comment:
>RegisterDTInterfaces_Callback [](start = 12, length = 29)

We should discuss more and I need to stare at code a bit more, but I *think* this entire function should run under the lock grabbed at beggining of the registerInterfaces.

The BeginClientCoreCallbackProcessing() that other callbacks grab will make sure that even if a callback arrives in middle of this function, it won't be processed until we're complete here.

The IoTHub layer *never* calls into a callback with a lock held, so there's not a deadlock happening here (where the DT.Lock() and calls into say invokeSetDevicetwin callback to do a IoTHub.Lock(), and then a callback would do a IoTHub.Lock() and a DT.Lock(); again per the IoTHub not locking on callbacks, this can't happen).

BeginClientCoreCallbackProcessing() is specifically designed to run in callbacks, not in the user thread.